marlowe-playground-client/src/MainFrame.purs

```diff
@@ -386,10 +384,123 @@ showErrorDescription (ConnectionError err) = "ConnectionError: " <> err
 
 runAjax ::
   forall m a.
-  ExceptT AjaxError (HalogenM FrontendState HAction ChildSlots Void m) a ->
-  HalogenM FrontendState HAction ChildSlots Void m (WebData a)
+  ExceptT AjaxError (HalogenM FrontendState Action ChildSlots Void m) a ->
+  HalogenM FrontendState Action ChildSlots Void m (WebData a)
 runAjax action = RemoteData.fromEither <$> runExceptT action
 
+------------------------------------------------------------
+handleGistAction ::
+  forall m.
+  MonadAff m =>
+  MonadEffect m =>
+  SPSettings_ SPParams_ -> GistAction -> HalogenM FrontendState Action ChildSlots Void m Unit
+handleGistAction settings PublishGist = do
+  description <- use (_newProject <<< NewProject._projectName)
+  let
+    playground = "{}"
+  simulation <- use (_simulationState <<< _marloweState)
+  currentSimulation <- toSimulation Simulation.editorGetValue
+  oldSimulation <- use (_simulationState <<< ST._oldContract)
+  -- soon we will separate the marlowe editor and the simulation, currently they are the same
+  marlowe <- toSimulation Simulation.editorGetValue
+  haskell <- HaskellEditor.editorGetValue
+  blockly <- query _blocklySlot unit (Blockly.GetWorkspace identity)
+  javascript <- query _jsEditorSlot unit (Monaco.GetText identity)
+  actus <- query _actusBlocklySlot unit (ActusBlockly.GetWorkspace identity)
+  let
+    files = { playground, currentSimulation, oldSimulation, simulation, marlowe, haskell, blockly, javascript, actus }
+
+    newGist = mkNewGist description files
+  void
+    $ runMaybeT do
+        -- TODO: gist state is duplicated until SCP-1256
+        mGist <- use _createGistResult
+        assign _createGistResult Loading
+        assign (_simulationState <<< ST._createGistResult) Loading
+        newResult <-
+          lift
+            $ case preview (_Success <<< gistId) mGist of
+                Nothing -> runAjax $ flip runReaderT settings $ Server.postApiGists newGist
+                Just gistId -> runAjax $ flip runReaderT settings $ Server.patchApiGistsByGistId newGist gistId
+        assign _createGistResult newResult
+        assign (_simulationState <<< ST._createGistResult) newResult
+        gistId <- hoistMaybe $ preview (_Success <<< gistId <<< _GistId) newResult
+        assign _gistUrl (Just gistId)
+        assign _loadGistResult (Right NotAsked)
+        assign (_simulationState <<< ST._gistUrl) (Just gistId)
+        assign (_simulationState <<< ST._loadGistResult) (Right NotAsked)
```

Comment:
I think it would help readability to do something like:

``` purescript
modifying _simulationState $ do assign ST._gistUrl (Just gistId)
                                assign ST._loadGistResult (Right NotAsked)
                                ...
```

Not certain of the functions there, but you get the idea.

Reply:
Ditto this comment for several other places where you assign `_simulationState`.

Reply:
FYI the correct answer is no.

Reply:
Can we agree on "oh, all right"?